agent/engine/docker_task_engine.go

```diff
@@ -950,18 +950,18 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 
 	// Set the credentials for pull from ECR if necessary
 	if container.ShouldPullWithExecutionRole() {
-		executionCredentials, ok := engine.credentialsManager.GetTaskCredentials(task.GetExecutionCredentialsID())
+		iamCredentials, ok, credentialsType := engine.getPullImageIamCredentials(task, container)
 		if !ok {
-			seelog.Errorf("Task engine [%s]: unable to acquire ECR credentials for image %s for container %s",
-				task.Arn, container.Image, container.Name)
+			seelog.Errorf("Task engine [%s]: unable to acquire ECR credentials for image %s for container %s with %sExecutionCredentials",
+				task.Arn, container.Image, container.Name, credentialsType)
 			return dockerapi.DockerContainerMetadata{
 				Error: dockerapi.CannotPullECRContainerError{
 					FromError: errors.New("engine ecr credentials: not found"),
 				},
 			}
 		}
 
-		iamCredentials := executionCredentials.GetIAMRoleCredentials()
+		seelog.Infof("Set RegistryAuthCredentials with %sExecutionCredentials", credentialsType)
```

Comment:
can we also log task arn and container name for more context here?

Reply:
will do!